### PR TITLE
Utilize MessageWnd for MultiPlayerLobbyWnd chat

### LINKED ISSUE #1995: Use chat windows in multiplayer lobby instead own controls.
### Description
Due `MessageWnd` supports autocompletion, text formatting and i18n I suggest to remove own controls in multiplayer lobby in favor the chat window used in the game and button to hide and show it.

### PR DESCRIPTION
**Pull request by dbenage-cx**
Replaces chat controls in MultiPlayerLobbyWnd with MessageWnd

Resolves #1995
Toggling chat window open/shown reserved for [rework of lobby layout](http://freeorion.org/forum/viewtopic.php?f=6&t=10408).

### CODE REVIEW

**Review by o01eg — CHANGES_REQUESTED** [review]
Could it be fixed it will work with correct player name if the player was disconnected before other player gets his messages from the chat's history?

File: util/MultiplayerCommon.h [hunk 1]
```diff
@@ -389,7 +389,7 @@ struct FO_COMMON_API ChatHistoryEntity {
     //@}
 
     boost::posix_time::ptime m_timestamp;
-    std::string m_player_name;
+    int m_player_id;
```
[the remark this answers is not in the document]
Comment (dbenage-cx) [reply util/MultiplayerCommon.h]: Thanks for catching, restored name and added color to history.

**Comment by dbenage-cx** [comment]
Rebased

File: server/ServerFSM.cpp [hunk 2]
```diff
@@ -336,20 +336,6 @@ bool ServerFSM::EstablishPlayer(const PlayerConnectionPtr& player_connection,
...
-        {
-            std::vector<std::reference_wrapper<const ChatHistoryEntity>> chat_history;
-            for (const auto& elem : m_server.GetChatHistory()) {
-                chat_history.push_back(std::cref(elem));
-            }
-            if (chat_history.size() > 0) {
-                player_connection->SendMessage(ChatHistoryMessage(chat_history));
-            }
-        }
     }
```
Comment (o01eg) [inline server/ServerFSM.cpp]: What was wrong with sending history in other states? Now player could connect to `PlayingGame` state too.

File: util/MultiplayerCommon.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (o01eg) [reply util/MultiplayerCommon.h]: Thanks.

File: server/ServerFSM.cpp (hunk 2 above)
Comment (dbenage-cx) [reply server/ServerFSM.cpp]: Remained from using player id, dropped this commit.

File: UI/MultiplayerLobbyWnd.cpp [hunk 3]
```diff
@@ -843,32 +828,30 @@ void MultiPlayerLobbyWnd::DoLayout() {
     g_preview_ul = GG::Pt(ClientWidth() - PREVIEW_SZ.x - CONTROL_MARGIN - PREVIEW_MARGIN, GG::Y(CONTROL_MARGIN + PREVIEW_MARGIN));
     m_preview_image->SizeMove(g_preview_ul, g_preview_ul + PREVIEW_SZ);
 
-    x = CHAT_WIDTH + CONTROL_MARGIN;
-    GG::Y y = std::max(m_save_file_text->RelativeLowerRight().y, m_preview_image->RelativeLowerRight().y) + 7*CONTROL_MARGIN;
-
-    GG::Pt any_can_edit_ul(x, y);
+    GG::Pt any_can_edit_ul(m_preview_image->RelativeUpperLeft().x + CONTROL_MARGIN, m_preview_image->RelativeLowerRight().y + CONTROL_MARGIN);
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.cpp]: break over two lines... a bit long for one

File: UI/MultiplayerLobbyWnd.h [hunk 4]
```diff
@@ -35,9 +36,10 @@ class MultiPlayerLobbyWnd : public CUIWnd {
     void KeyPress(GG::Key key, std::uint32_t key_code_point, GG::Flags<GG::ModKey> mod_keys) override;
 
     void            ChatMessage(int player_id, const boost::posix_time::ptime& timestamp, const std::string& msg);
-    void            ChatMessage(const std::string& player_name,
-                                const boost::posix_time::ptime& timestamp,
-                                const std::string& msg);
+    void            ChatMessage(const std::string& text,
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.h]: `message_text` might be clearer?

File: util/SerializeMultiplayerCommon.cpp [hunk 5]
```diff
@@ -143,9 +143,10 @@ template void MultiplayerLobbyData::serialize<freeorion_xml_iarchive>(freeorion_
 template <class Archive>
 void ChatHistoryEntity::serialize(Archive& ar, const unsigned int version)
 {
-    ar  & BOOST_SERIALIZATION_NVP(m_timestamp)
+    ar  & BOOST_SERIALIZATION_NVP(m_text)
```
Comment (geoffthemedio) [inline util/SerializeMultiplayerCommon.cpp]: probably should add a version number and handle older versions when loading to preserve save compatibility

**Comment by geoffthemedio** [comment]
Probably should prevent the messages window from being dragged / manually resized...
messags_in_mp_lobby

**Comment by geoffthemedio** [comment]
@o01eg could you make the messages wnd not draggable?

**Comment by o01eg** [comment]
@geoffthemedio I'm not sure. I have a little experience with UI.
Looks like it should be either border-less message window or a simple panel with internals of message window.

**Comment by geoffthemedio** [comment]
I'd start with just making it not draggable. There's a simple flag for that.

**Pull request merged**